internal/server/http/middleware/health_check_test.go

```diff
@@ -0,0 +1,126 @@
+package middleware
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/middleware/healthcheck"
+	serviceMock "github.com/green-ecolution/green-ecolution-backend/internal/service/_mock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHealthCheck(t *testing.T) {
+	app := fiber.New()
+  svc := serviceMock.NewMockServicesInterface(t)
+  svc.EXPECT().AllServicesReady().Return(true)
+  handler := HealthCheck(svc)
+	app.Use(handler)
+
+	t.Run("should return 200 OK for liveness probe", func(t *testing.T) {
+		req := httptest.NewRequest(http.MethodGet, "/health", nil)
+		resp, err := app.Test(req, -1)
+
+		assert.NoError(t, err)
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+	})
+
+	t.Run("should return 200 OK for readiness probe when services are ready", func(t *testing.T) {
+		req := httptest.NewRequest(http.MethodGet, "/ready", nil)
+		resp, err := app.Test(req, -1)
+
+		assert.NoError(t, err)
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+	})
+
+	t.Run("should return 404 for undefined endpoint", func(t *testing.T) {
+		req := httptest.NewRequest(http.MethodGet, "/undefined", nil)
+		resp, err := app.Test(req, -1)
+
+		assert.NoError(t, err)
+		assert.Equal(t, http.StatusNotFound, resp.StatusCode)
+	})
+}
+
+func TestHealthCheck_EdgeCases(t *testing.T) {
```

Comment:
I think we don't need that either, these are all edge cases that we don't need to test on the side of this middleware.